### PR TITLE
Upgraded pcodec to 0.2 and used new ModeSpec configuration

### PR DESCRIPTION
**Pull request by mwlon**
Upgraded Pcodec to 0.2. The main API change is replacement of individual mode specs with a single spec for all modes.

### CODE REVIEW

**Comment by mwlon** [comment]
It seems the lines the coverage report is complaining about are deliberately unreachable (with correct mypy typing)

File: numcodecs/pcodec.py [hunk 1]
```diff
@@ -60,20 +57,25 @@ def __init__(
...
     def encode(self, buf):
         buf = ensure_contiguous_ndarray(buf)
 
+        match self.mode_spec:
+            case 'auto':
+                mode_spec = ModeSpec.auto()
+            case 'classic':
+                mode_spec = ModeSpec.classic()
+            case _:
+                raise ValueError(f"unknown value for mode_spec: {self.mode_spec}")
```
Comment (rabernat) [inline numcodecs/pcodec.py:72]:
```suggestion
            case _:  # pragma: no cover
                raise ValueError(f"unknown value for mode_spec: {self.mode_spec}")
```

**Review by rabernat — APPROVED** [review]
Thanks for doing this @mwlon!

My suggestion should fix the coverage.

**Comment by mwlon** [comment]
Now this is confusing

**Comment by mwlon** [comment]
Oh it looks like it's failing due to unrelated changes from the merge (something zfpy). And I can't tell why ruff is complaining with this last commit... it passes locally, and the "Details" page doesn't say what file it's failing on. @rabernat @normanrz are we good to merge this?

**Comment by rabernat** [comment]
Thanks for your patience with our over-eager CI @mwlon! 🙏 

Yes, you're right that #540 broke our coverage test, so we can ignore that. I wish we could fix pre-commit though. I'll see what I can do.

**Pull request merged**